runtime/ops/spawn.rs

```diff
@@ -226,17 +233,18 @@ async fn op_spawn_wait(
   state: Rc<RefCell<OpState>>,
   rid: ResourceId,
 ) -> Result<ChildStatus, AnyError> {
+  #![allow(clippy::await_holding_refcell_ref)]
   let resource = state
     .borrow_mut()
     .resource_table
-    .take::<ChildResource>(rid)?;
-  Rc::try_unwrap(resource)
-    .ok()
-    .unwrap()
-    .0
-    .wait()
-    .await?
-    .try_into()
+    .get::<ChildResource>(rid)?;
+  let result = resource.0.borrow_mut().wait().await?.try_into();
+  state
+    .borrow_mut()
+    .resource_table
+    .close(rid)
+    .expect("shouldn't have closed until now");
+  result
```

Comment:
The spawn code change looks good, but I don't like this refcell being hold across await points. Can you instead try to borrow, and if it fails return a `Deno.errors.Busy` error?

Reply:
> Can you instead try to borrow, and if it fails return a `Deno.errors.Busy` error?

Could you clarify this suggestion? It seems like the content of the refcell is needed along the `.wait()` call so this is unavoidable. For context this is the only place it's borrowed, and the only reason I need a refcell is to wrestle with the other reference in the resource table (which didn't previously exist because it was taken out on wait start).

Reply:
Ok, ignore the first part of my first comment. The issue is that this code will panic if you call this op concurrenly.

To avoid this, you must switch the `borrow_mut()` on line 241 to a `try_borrow_mut()` and error when that returns `Err(BorrowMutError)`.